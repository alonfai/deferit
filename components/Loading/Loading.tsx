import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { EvilIcons } from '@expo/vector-icons';

export type Props = {};

/**
 * Render a Loading data component when data is been refetched
 */
const Loading: React.FC<Props> = () => {
  return (
    <View style={styles.loading}>
      <Text style={styles.loadingText}>Loading Data...</Text>
      <View>
        <EvilIcons name='spinner' size={24} color='black' />
      </View>
    </View>
  );
};
const styles = StyleSheet.create({
  loading: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  loadingText: {
    fontSize: 20,
    marginRight: 5,
  },
});

export default Loading;
