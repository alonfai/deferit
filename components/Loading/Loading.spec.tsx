/// <reference types="@testing-library/jest-native" />
import React from 'react';
import { render } from '@testing-library/react-native';

import Loading from './Loading';

describe('<Loading />', () => {
  test('Render successfully', async () => {
    const { getByText, toJSON, findByA11yLabel } = render(<Loading />);
    const loadingText = getByText('Loading Data...');
    const icon = findByA11yLabel('spinner');
    expect(loadingText).toBeTruthy();
    expect(icon).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
});
