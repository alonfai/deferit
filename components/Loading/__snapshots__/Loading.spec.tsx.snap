// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> Render successfully 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "marginRight": 5,
      }
    }
  >
    Loading Data...
  </Text>
  <View>
    <Text />
  </View>
</View>
`;
